core/src/main/java/com/datastax/oss/driver/api/core/cql/Statement.java

```diff
@@ -165,6 +166,20 @@ default T setRoutingKey(ByteBuffer... newRoutingKeyComponents) {
    */
   T setPagingState(ByteBuffer newPagingState);
 
+  /**
+   * Calculates the approximate size in bytes that the statement will have when encoded.
+   *
+   * <p>The size might be over-estimated by a few bytes due to global options that may be defined on
+   * a {@link Session} but not explicitly set on the statement itself.
+   *
+   * <p>The result of this method is not cached, calling it will cause some encoding to be done in
+   * order to determine some of the statement's attributes sizes. Therefore, use this method
+   * sparingly in order to avoid unnecessary computation.
+   *
+   * @return the approximate number of bytes this statement will take when encoded.
+   */
+  int computeSizeInBytes(DriverContext context);
```

Comment:
The 3.x implementation used the name `requestSizeInBytes` for this method, where we use `computeSizeInBytes`  here.  I think I like the new name better (since `compute` seems more explicit), but just wanted to point out they were different.